File: frontend/src/backend/shoppingService.js
import {appFetch} from './appFetch.js';

export const buy = async (sessionId, quantity, creditCardNumber) =>
    await appFetch('POST', '/orders/buy', {sessionId, quantity, creditCardNumber});

export const findOrders = async ({page}) =>
    await appFetch('GET', `/orders?page=${page}`);

export const deliverTickets = async (orderId, creditCardNumber) =>
    await appFetch('POST', `/orders/${orderId}/deliver`, {creditCardNumber});

File: backend/src/main/java/es/udc/paproject/backend/model/services/ShoppingService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Order;
import es.udc.paproject.backend.model.exceptions.CollectedTicketsException;
import es.udc.paproject.backend.model.exceptions.IncorrectCreditCardException;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.LocalitiesExceededException;
import es.udc.paproject.backend.model.exceptions.MovieAlreadyStartedException;
import es.udc.paproject.backend.model.exceptions.PermissionException;

public interface ShoppingService {

    Order buy(Long userId, Long sessionId, int quantity, String creditCardNumber)
        throws InstanceNotFoundException, LocalitiesExceededException, MovieAlreadyStartedException;

        void deliverTickets(Long orderId, String creditCardNumber)
            throws InstanceNotFoundException, CollectedTicketsException, IncorrectCreditCardException,
            MovieAlreadyStartedException;

    Block<Order> findOrders(Long userId, int page, int size);
}
